### PR TITLE
BAH-3375 | Add. Rest Client Implementation for  Create Lab Test, Lab Panel and Radiology Test

### PR DESCRIPTION
**Pull request by parvathy00**
Jira -> [BAH-3375](https://bahmni.atlassian.net/browse/BAH-3375)

In this PR, we implement the REST JSON Based Strategy for Create Lab Tests, Lab Panels and Radiology Tests.

### CODE REVIEW

File: openerp-atomfeed-service/src/main/resources/jobs-context.xml [hunk 1]
```diff
@@ -200,8 +245,12 @@
                     <ref bean="openelisSaleOrderFeedFailedJobTrigger" />
                     <ref bean="eventRecordsNumberOffsetMarkerTrigger"/>
                     <ref bean="eventPublisherTrigger"/>
-                    <ref bean="openerpLabFeedJobTrigger" />
-                    <ref bean="openerpLabFeedFailedJobTrigger" />
+                    <ref bean="openerpLabTestFeedJobTrigger" />
```
Comment (gsluthra) [inline openerp-atomfeed-service/src/main/resources/jobs-context.xml:248]: Please sequence the line as NormalEventA,FailedEventA, NormalEventB, FailedEventB, etc.

File: openerp-atomfeed-service/src/main/java/org/bahmni/feed/openerp/OpenERPAtomFeedProperties.java [hunk 2]
```diff
@@ -71,9 +73,11 @@ public String getOdooURIForJob(Jobs feedJob, boolean isRestEnabled) {
         if (isRestEnabled) {
```
Comment (gsluthra) [inline openerp-atomfeed-service/src/main/java/org/bahmni/feed/openerp/OpenERPAtomFeedProperties.java:73]: I would have created two different methods instead called: getOdooXMLUriEndpoint() and getOdooRESTUriEndPoint(forFeed). The caller already knows whether its XML or Rest.. why keep passing the flag and put this logic here? Any advantage?

File: openerp-atomfeed-service/src/main/resources/jobs-context.xml (hunk 1 above)
Comment (parvathy00) [reply openerp-atomfeed-service/src/main/resources/jobs-context.xml:248]: Sure, will do that.

File: openerp-atomfeed-service/src/main/java/org/bahmni/feed/openerp/OpenERPAtomFeedProperties.java (hunk 2 above)
Comment (parvathy00) [reply openerp-atomfeed-service/src/main/java/org/bahmni/feed/openerp/OpenERPAtomFeedProperties.java:73]: Makes sense, will do.

File: openerp-atomfeed-service/src/main/java/org/bahmni/feed/openerp/worker/OpenERPLabPanelServiceEventWorker.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        try {
+            OpenMRSLabOrderTypeEvent openMRSLabOrderTypeEvent = labOrderTypeEventMap.get(event.getTitle());
+            if(openMRSLabOrderTypeEvent == null) return ;
+            openERPContext.execute(mapRequest(event, openMRSLabOrderTypeEvent), odooURL);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private OpenERPRequest mapRequest(Event event, OpenMRSLabOrderTypeEvent openMRSLabOrderTypeEvent) throws IOException {
```
[the remark this answers is not in the document]
Comment (gsluthra) [reply openerp-atomfeed-service/src/main/java/org/bahmni/feed/openerp/worker/OpenERPLabPanelServiceEventWorker.java:50]: Then why does the code say "Orders"?

File: openerp-atomfeed-service/src/main/java/org/bahmni/feed/openerp/worker/OpenERPLabPanelServiceEventWorker.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        this.webClient = openMRSWebClient;
+        this.urlPrefix = urlPrefix;
+        labOrderTypeEventMap.put(OpenMRSLabPanelEvent.LAB_PANEL_EVENT_NAME, new OpenMRSLabPanelEvent());
+    }
+
+    @Override
+    public void process(Event event) {
+        logger.debug("Processing the event [{}]", event.getContent());
+        try {
+            OpenMRSLabOrderTypeEvent openMRSLabOrderTypeEvent = labOrderTypeEventMap.get(event.getTitle());
```
[the remark this answers is not in the document]
Comment (gsluthra) [reply openerp-atomfeed-service/src/main/java/org/bahmni/feed/openerp/worker/OpenERPLabPanelServiceEventWorker.java:42]: okay. i see.

File: openerp-atomfeed-service/src/main/java/org/bahmni/feed/openerp/worker/OpenERPLabPanelServiceEventWorker.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (parvathy00) [reply openerp-atomfeed-service/src/main/java/org/bahmni/feed/openerp/worker/OpenERPLabPanelServiceEventWorker.java:50]: Yeah, in that case these names seems to be misleading. This is the existing code, no modifications have been made to these class names. Currently, there are classes named OpenMRSLabTest, OpenMRSLabPanel, and OpenMRSRadiologyTest, all defined as different OpenMRSLabOrderTypes.

[the remark this answers is not in the document]
Comment (gsluthra) [reply openerp-atomfeed-service/src/main/java/org/bahmni/feed/openerp/worker/OpenERPLabPanelServiceEventWorker.java:50]: Ok. will approve for now. Plz add a comment on this code to indicate name is a bit misleading of this class here.

**Review by gsluthra — APPROVED** [review]

**Pull request merged**